bst.py

```diff
@@ -190,52 +208,186 @@ def breadth_first_traversal(self):
             if leaf.right:
                 q.put(leaf.right)
 
+    def get_dot(self):
+        """return the tree with root 'self' as a dot graph for visualization"""
+        return "digraph G{\n%s}" % ("" if self.root.key is None else (
+            "\t%s;\n%s\n" % (
+                self.root.key,
+                "\n".join(self.root._get_dot())
+            )
+        ))
+
+    def graph(self, file_name):
+        t = subprocess.Popen(["dot", "-Tpng"],
+                             stdin=subprocess.PIPE,
+                             stdout=subprocess.PIPE)
+        png_data = t.communicate(self.get_dot())
+        with open('{}.png'.format(file_name), 'w') as png_file:
+            png_file.write(png_data[0])
+        os.system('open {}.png'.format(file_name))
+
+    def nuclear_option(self):
+        generator = self.in_order()
+        num = []
+        while True:
+            try:
+                num.append(generator.next())
```

Comment:
It's probably easier if you're going to cast the generator to a list just to do it:

```python
num = list(generator)
```